userspace/falco/app/actions/load_config.cpp

```diff
@@ -18,15 +18,42 @@ limitations under the License.
 #include "actions.h"
 #include "falco_utils.h"
 
+/* DEPRECATED: we will remove it in Falco 0.38. */
+#define FALCO_BPF_ENV_VARIABLE "FALCO_BPF_PROBE"
+
 using namespace falco::app;
 using namespace falco::app::actions;
 
-// applies legacy/in-deprecation options to the current config
-static void apply_deprecated_options(
-		const falco::app::options& opts,
-		const std::shared_ptr<falco_configuration>& cfg)
+// applies legacy/in-deprecation options to the current state
+static falco::app::run_result apply_deprecated_options(falco::app::state& s)
 {
-	// Keep for future use cases.
+	// If overridden from CLI options (soon to be removed),
```

Comment:
Here we are merging CLI options into relevant config structures.
This way, we can trust `config` values.